### PR TITLE
cmd/data: teach Data how to Marhsal()

### PR DESCRIPTION
**Pull request by ttaylorr**

### CODE REVIEW

File: cmd/data/data.go [hunk 1]
```diff
@@ -56,17 +53,23 @@ func (d *data) Read(c *chunk.Chunk) error {
...
+// Payload represents the actual data encoded in each Data frame.
+func (d *data) Payload() []byte { return d.data[1:] }
+
 // Marshal implements the Data.Marshal, using the same header that was sent
 // during the original read.
 func (d *data) Marshal() (*chunk.Chunk, error) {
 	return &chunk.Chunk{
 		Header: d.header,
-		Data:   append([]byte{d.Control}, d.Payload...),
+		Data:   d.data,
```
Comment (ttaylorr) [inline cmd/data/data.go]: woot

File: cmd/data/stream.go [hunk 2]
```diff
@@ -44,6 +52,10 @@ func (s *Stream) Chunks() chan<- *chunk.Chunk { return s.chunks }
 // parsed from the RTMP chunk stream on which this `*data.Stream` is listening.
 func (s *Stream) In() <-chan Data { return s.in }
 
+// Out returns a write-only channel that, when written to, sends data back to
+// the client.
+func (s *Stream) Out() chan<- Data { return s.out }
```
Comment (connor4312) [inline cmd/data/stream.go]: The semantics of In() and Out() seem to be reversed here. Usually you write _in_ to the read channel and read _out_ of the out channel

**Comment by connor4312** [comment]
Works for me

**Comment by Doridian** [comment]
lgtm

File: cmd/data/stream.go (hunk 2 above)
Comment (ttaylorr) [inline cmd/data/stream.go]: I agree. Fixed in f938a35d4ddf8d480a059cfb8485d45e1060a59a.

**Comment by ttaylorr** [comment]
Thanks for the review @Doridian, @connor4312 👋

Going to wait for Travis to finish this build and then merge away. 😄

**Pull request merged**